app/MapScreen.js

```diff
@@ -65,16 +199,122 @@ export default class MapScreen extends Component {
     }
     tileQueryParams['eo:bands'] = bandCombinations[bandType];
 
+    const tileQueryParamsUI = {
+      bandCombination: bandType,
+      date: endDate,
+      cloudPercentage: lexSlotValues.CloudPercentage || 0,
+    };
+
     this.setState({
+      tileQueryParamsUI,
       tileQueryString: queryString.stringify(tileQueryParams),
     });
   }
 
+  finishRecording(filePath) {
+    this.setState({
+      isRecording: false,
+    });
+    console.log(`Finished recording at path: ${filePath}`);
+  }
+
+  prepareRecorder() {
+    const audioPath = `${AudioUtils.DocumentDirectoryPath}/test2.lpcm`;
+    AudioRecorder.prepareRecordingAtPath(audioPath, {
+      SampleRate: 8000,
+      Channels: 1,
+      AudioQuality: 'High',
+      AudioEncoding: 'lpcm',
+      IncludeBase64: true,
+    });
+
+    AudioRecorder.onFinished = this.onAudioRecordingFinished;
+  }
+
+  prepareRecordingAnimation() {
+    const { animatedShadowRadius } = this.state;
+
+    const animatedRadiusValues = [5, 10, 15, 8, 12, 18, 10, 7];
+    const animations = animatedRadiusValues.map(radiusValue => Animated.timing(
+      animatedShadowRadius,
+      {
+        toValue: radiusValue,
+        duration: 200,
+        useNativeDriver: true,
+      },
+    ));
+
+    this.recordingAnimation = Animated.loop(Animated.sequence(animations));
+  }
+
+  updateMap(feature, lexSlotValues) {
+    const { tileQueryParamsUI } = this.state;
+
+    this.convertLexSlotsToQueryParams(lexSlotValues);
+
+    this.setState({
+      centerCoords: feature.geometry.coordinates,
+    });
+  }
+
+  async startRecording() {
+    const { isAuthorized } = this.state;
+    if (!isAuthorized) {
+      return;
+    }
+
+    this.recordingAnimation.start();
+
+    this.setState({
+      isRecording: true,
+      errorMessage: null,
+    });
+
+    try {
+      await AudioRecorder.startRecording();
+    } catch (error) {
+      console.error(error);
```

Comment:
Do we need to handle these errors and present anything to the screen? Just in case recording fails to start/stop

Reply:
Good idea, I'll update that.